network/handlers/drain.go

```diff
@@ -79,6 +86,10 @@ func (d *Drainer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if d.draining() {
 			http.Error(w, "shutting down", http.StatusServiceUnavailable)
 		} else {
+			if d.CheckOption != nil {
+				d.CheckOption(w, r)
+				return
+			}
```

Comment:
```suggestion
		} else  if d.CheckOption != nil {
				d.CheckOption(w, r)
	         } else {
	                         w.WriteHeader(http.StatusOK)
	         }         
```

Reply:
done

Reply:
It seems after changing to this, Unit test TestMetricsExport/OpenCensus starts to fail and downstream unit test in serving fails as well.

Reply:
After retesting, all tests passed. Not sure if the failure is just a coincidence. I'll rerun the test for more times to verify.